action/src/plugins/staticHosting/plugin.ts

```diff
@@ -29,18 +29,27 @@ const runStaticHostingPlugin = async (
     return;
   }
 
+  const isFile = fs.lstatSync(pathToUpload).isFile();
+
   // get signed url
   const { signedUrl, fields, objectPath, hostingUrl } = await createSignedUrl({
     ghOwner: owner,
     ghRepo: repo,
     ghSha,
     ghToken,
-    taskId
+    taskId,
+    filename: isFile ? path.basename(pathToUpload) : undefined
   });
 
-  // upload directory
-  core.info(`[${taskId}] Uploading ${pathToUpload} to ${objectPath}...`);
-  await uploadDirectory(signedUrl, fields, pathToUpload, objectPath);
+  if (isFile) {
+    // upload file
+    core.info(`[${taskId}] Uploading file ${pathToUpload} to ${objectPath}...`);
```

Comment:
The `uploadFileWithSignedUrl` already logs this. So the log here can be removed.